gravitee-am-cypress/cypress/integration/oidc/ciba/ciba.spec.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+    createAcceptAllDeviceNotifier,
+    createDomain,
+    createUser,
+    requestAdminAccessToken,
+    requestDomain, startDomain
+} from "../../utils/management-utils";
+import {domainOidcCibaConfig, oidcApplication} from "../../../fixtures/oidc";
+import {User} from "../../../model/users";
+
+const configureDomainForCiba = (domain, accessToken, method, body) =>
+    requestDomain(domain, accessToken, "PATCH", body);
+
+const createCibaApplication = (domainHrid) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/register`,
+    method: 'POST',
+    retryOnStatusCodeFailure: true,
+    body: oidcApplication
+});
+
+const registerDomainWithNotifier = (domain, domainHrid, clientId, clientSecret) => cy.request({
+    url: Cypress.env().cibaDelegateNotifierUrl + "/domains",
+    method: 'POST',
+    body: {
+        domainId: domain,
+        domainCallback: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate/callback`,
+        clientId: clientId,
+        clientSecret: clientSecret
+    }
+})
+
+const requestAuthentication = (domainHrid, clientId, clientSecret, loginHint, userCode) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate`,
+    method: 'POST',
+    auth: {
+        user: clientId,
+        password: clientSecret
+    },
+    form: true,
+    body: {
+        scope: "openid",
+        login_hint: loginHint,
+        user_code: userCode
+    },
+    failOnStatusCode: false
+})
```

Comment:
I would suggest keeping these types of functions out of the spec file, especially as the functions mentioned here are general in nature and can be reused in other spec files.
In APIM we have a _commands_ folder directly under the _cypress_ folder to store those functions. Could be a good idea to keep it the same across products.

Reply:
These are specific to CIBA so I don't think it would be a good strategy to keep everything under one directory. I think these can be moved to a separate file but within the same `ciba` directory.

Reply:
That does apply to the management requests, however. Cheers.